mne/annotations.py

```diff
@@ -517,3 +517,119 @@ def _ensure_annotation_object(obj):
     if not isinstance(obj, Annotations):
         raise ValueError('Annotations must be an instance of '
                          'mne.Annotations. Got %s.' % obj)
+
+
+def _counter_factory():
+    """Create a callable counter that takes 1 parameter.
+
+    It can be used as follows::
+
+        >>> my_counter = _counter_factory()
+        >>> my_counter(None)
+        1
+        >>> my_counter('usless string')
+        2
+        >>> my_counter(42)
+        3
+        >>> [my_counter(x) for x in ['a', None, 42]]
+        [4, 5, 6]
+        >>> my_other_counter = _counter_factory()
+        >>> [my_other_counter(x) for x in ['a', None, 42]]
+        [1, 2, 3]
+
+    """
+    class Counter():
+        count = 1
+
+        def __call__(self, *args, **kargs):
+            c = self.count
+            self.count += 1
+            return c
+
+    return Counter()
+
+
+@verbose
+def events_from_annotations(raw, event_id=None, regexp=None,
+                            event_id_func=None, verbose=None):
+    """Get events and event_id from an Annotations object.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data for which Annotations are defined.
+
```

Comment:
No need for this blank line. We don't do it elsewhere unless required by the ReStructuredText that precedes it.